src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/server42/Server42YFlowRttSpec.groovy

```diff
@@ -0,0 +1,672 @@
+                 assert subFlow1Stats.get(FLOW_RTT, REVERSE, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+            }
+        }
+
+        and: "Stats for FORWARD and REVERSE directions are available for the second sub-flow"
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT, 1) {
+            verifyAll(flowStats.of(yFlow.subFlows.last().flowId)) { subFlow2Stats ->
+                assert subFlow2Stats.get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+                assert subFlow2Stats.get(FLOW_RTT, REVERSE, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+            }
+        }
+
+        when: "Disable switch rtt toggle on ep1 and ep2 ends"
+        switchHelper.setServer42FlowRttForSwitch(swT.ep1, false, true)
+        switchHelper.setServer42FlowRttForSwitch(swT.ep2, false, true)
+        checkpointTime = new Date().getTime()
+
+        then: "Stats for FORWARD direction are available for both sub-flows"
+        Wrappers.wait(STATS_FROM_SERVER42_LOGGING_TIMEOUT + WAIT_OFFSET, 1) {
+            assert flowStats.of(yFlow.subFlows.first().flowId).get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+            assert flowStats.of(yFlow.subFlows.last().flowId).get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+        }
+
+        when: "Disable global toggle"
+        northbound.toggleFeature(FeatureTogglesDto.builder().server42FlowRtt(false).build())
+        switchHelper.waitForS42SwRulesSetup(false)
+
+        and: "Wait for several seconds"
+        checkpointTime = new Date().getTime()
+
+        then: "Expect no flow rtt stats for FORWARD and REVERSE direction for the first sub-flow"
+        verifyAll(flowStats.of(yFlow.subFlows.first().flowId)) { subFlow1Stats ->
+            assert !subFlow1Stats.get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+            assert !subFlow1Stats.get(FLOW_RTT, REVERSE, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+        }
+
+        and: "Expect no flow rtt stats for FORWARD and REVERSE direction for the second sub-flow"
+        verifyAll(flowStats.of(yFlow.subFlows.last().flowId)) { subFlow2Stats ->
+            !subFlow2Stats.get(FLOW_RTT, FORWARD, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+            !subFlow2Stats.get(FLOW_RTT, REVERSE, SERVER_42).hasNonZeroValuesAfter(checkpointTime)
+        }
+
+        cleanup: "Revert system to original state"
+        yFlow && yFlow.delete() && switchHelper.verifyAbsenceOfServer42FlowRttRules(initialSwitchesProps.keySet())
+        flowRttToggleInitialState != null && northbound.toggleFeature(FeatureTogglesDto.builder().server42FlowRtt(flowRttToggleInitialState).build())
+        initialSwitchesProps && switchHelper.revertToOriginSwitchSetup(initialSwitchesProps, flowRttToggleInitialState)
+
+        where:
+        isShEpYPoint << [true, false]
```

Comment:
Again, this variable name looks like a generated password.

Reply:
updated